src/components/ExperienceDetail/ReportModal.js

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import R from 'ramda';
+import { useParams } from 'react-router-dom';
+import Modal from 'common/Modal';
+import usePermission from 'hooks/usePermission';
+import ApiErrorFeedback from './ReportForm/ApiErrorFeedback';
+import ReportSuccessFeedback from './ReportForm/ReportSuccessFeedback';
+import { queryExperienceIfUnfetched } from 'actions/experience';
+import ReportForm from './ReportForm';
+import { MODAL_TYPE } from './ReportForm/constants';
+import PropTypes from 'prop-types';
+
+// from params
+const experienceIdSelector = R.prop('id');
+const useExperienceId = () => {
+  const params = useParams();
+  return experienceIdSelector(params);
+};
+
+const ReportModal = ({
+  modalState,
+  handleIsModalOpen,
+  closableOnClickOutside,
+  setModalClosableOnClickOutside,
+}) => {
+  const { isModalOpen, modalType, modalPayload } = modalState;
+  const experienceId = useExperienceId();
+  const [, fetchPermission] = usePermission({
+    publishId: experienceId,
+  });
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(queryExperienceIfUnfetched(experienceId));
+  }, [dispatch, experienceId]);
+
+  useEffect(() => {
+    fetchPermission();
+  }, [experienceId, fetchPermission]);
+
+  const renderModalChildren = useCallback(
```

Comment:
Suggestion: We can make it a pure functional component.

```js
const ModalContent = ({ modalState ... }) => {
...
}
```
so as to avoid nested components in useCallback.

Reply:
I made it to a pure func component. Thanks for suggestion.